feature_engine/datetime/datetime_subtraction.py

```diff
@@ -0,0 +1,350 @@
+                    "Number of items in new_variables_names must be equal to number of "
+                    "items in reference_variables * items in variables to "
+                    "combine. In other words, "
+                    "the transformer needs as many new variable names as reference "
+                    "variables to perform over the variables to "
+                    "combine."
+                )
+
+        if missing_values not in ["raise", "ignore"]:
+            raise ValueError("missing_values takes only values 'raise' or 'ignore'")
+
+        if not isinstance(drop_original, bool):
+            raise TypeError(
+                "drop_original takes only boolean values True and False. "
+                f"Got {drop_original} instead."
+            )
+
+        if not isinstance(dedupe_variable_pairs, bool):
+            raise TypeError(
+                "dedupe_variable_pairs takes only boolean values True and False. "
+                f"Got {dedupe_variable_pairs} instead."
+            )
+
+        self.reference_variables = reference_variables
+        self.variables_to_combine = variables_to_combine
+        self.output_unit = output_unit
+        self.dedupe_variable_pairs = dedupe_variable_pairs
+        self.new_variables_names = new_variables_names
+        self.missing_values = missing_values
+        self.drop_original = drop_original
+        self.dayfirst = dayfirst
+        self.yearfirst = yearfirst
+        self.utc = utc
+
+    def fit(self, X: pd.DataFrame, y: Optional[pd.Series] = None):
+        """
+        This transformer does not learn any parameter.
+
+        Parameters
+        ----------
+        X: pandas dataframe of shape = [n_samples, n_features]
+            The training input samples. Can be the entire dataframe, not just the
+            variables to transform.
+
+        y: pandas Series, or np.array. Default=None.
+            It is not needed in this transformer. You can pass y or None.
+        """
+
+        # check input dataframe
+        X = _is_dataframe(X)
+
+        # find or check for datetime variables to combine
+        self.variables_to_combine_ = _find_or_check_datetime_variables(
+            X, self.variables_to_combine
+        )
+
+        # find or check for datetime reference variables
+        self.reference_variables_ = _find_or_check_datetime_variables(
+            X, self.reference_variables
+        )
+
+        # check if dataset contains na
+        if self.missing_values == "raise":
+            _check_contains_na(X, self.reference_variables_)
+            _check_contains_na(X, self.variables_to_combine_)
+
+            _check_contains_inf(X, self.reference_variables_)
+            _check_contains_inf(X, self.variables_to_combine_)
+
+        variable_pairs = list(itertools.product(self.reference_variables_,
+                                                self.variables_to_combine_))
+
+        if self.dedupe_variable_pairs:
```

Comment:
I think this should be a default functionality and not optional. After all, what is the use of subtracting one variable from itself? it would be just 0s. Unless you have a specific example of when this would be useful?

Reply:
it's removing the 0s and the instances of v1 - v2 and v2 - v1. see below

Reply:
I agree that removing 0s could be the default, but is the user going to want the option for second case?